### DISCUSSION TITLE
Does micropython for Pico W support PEAP/MSCHAPV2 ?

### DISCUSSION THREAD

**Discussion by jjzheng01** [micropython/micropython#10399] [Using MicroPython]
Hi, I want to connect the Pico W to WiFi network which requires the username and password. But there is no document about it.
Does the micropython for the Pico W support PEAP/MSCHAPV2 ? Thanks.

I have tried this in Thonny:
https://docs.pycom.io/tutorials/networks/wlan/#connecting-to-a-wpa2-enterprise-network
Connecting with EAP-PEAP or EAP-TTLS: In case of EAP-PEAP (or EAP-TTLS), the client key and certificate are not necessary, only a username and password pair. If it is required to validate the server’s public key, an appropriate CA certificate (chain) must also be provided.

```py
from network
wlan = network.WLAN(network.STA_IF)
wlan.active(True)
# this
wlan.connect(ssid='my_wifi', auth=(network.WLAN.WPA2_ENT, 'my_username', 'my_password'))
# or this
wlan.connect(ssid='my_wifi', auth=(network.WLAN.MSCHAPV2, 'my_username', 'my_password'))
```
The errors are
AttributeError: type object 'CYW43' has no attribute 'WPA2_ENT'
or
AttributeError: type object 'CYW43' has no attribute 'MSCHAPV2'

**Comment by jimmo** [comment]
@jjzheng01 Not yet. The chip supports it, it's just a matter of adding driver support.

The docs you're referring to are for the (now-defunct) Pycom fork of MicroPython.

**Reply by jjzheng01** [reply]
@jimmo Thank you for your information and clarification. 
It is a pity that this feature is not supported yet. I'll see if I can add the driver. I don't know what effort I should put in to get it done in this moment.

**Comment by SL2021-Dev** [comment]
I have been requesting this since version 1.17.
Is there any discussion among the maintainers to implement this?
From what I understand all the pieces are there, but I am not proficient enough when it comes to make files.
Please consider adding this soon !!!!!!

https://github.com/orgs/micropython/discussions/12143

**Comment by jouellnyc** [comment]
This is workaround / possible solution using a Raspberry Pi Zero. ***Please see the many (including security) warnings***:

https://github.com/jouellnyc/wpa-ent-mschapv2-bridge

Keep in mind your mileage may vary. I did the best I could at the time to get something working in a lab environment.
